R/validateFile.R

```diff
@@ -1,6 +1,6 @@
 validateFile <- function(db, fileName, ...) {
 
-  on.exit(try({close(srcFile); close(destFile)}, silent = TRUE))
+  on.exit(try({close(srcFile); close(destFile); unlink(db1$src)}, silent = TRUE))
```

Comment:
Can we have a test case for this which will currently fail? I guess the original logic was that R and the OS will clean up temp files when the process exits. Is this not the case?

Reply:
We are encountering this error regularly for some time now

```
ERROR [2023-09-20 12:55:36] Escalated warning: '/tmp/RtmpSIgbgj/gulawkje' already exists
Error in dir.create(db1$src) : 
  Escalated warning: '/tmp/RtmpSIgbgj/gulawkje' already exists
```

`validate.file` does not create an actual temp file, but rather a file in with a random file name in tempdir.

```
paste0(tempdir(), "/", paste0(sample(letters, 8), collapse = ""), "/")
```

Reply:
I have added a test case and some documentation

Reply:
Thanks. Fine for me then.

I don't really understand how we end up with the same file names. Is it because we set the seed in those programs?